### PR TITLE
Enabled service to invoke another AWS Lambda function.

### PR DESCRIPTION
**Pull request by pouncilt**

### CODE REVIEW

File: chucks-pick3-ws/src/main/java/software/bytepushers/pick3/services/Pick3PredcitionServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import org.springframework.stereotype.Service;
+
+@Service
+public class Pick3PredcitionServiceImpl implements Pick3PredictionService {
+
+    @Override
+    public int[][] generatePredictions(int winningNumber) {
+        final Pick3PredictionService pick3PredictionService = LambdaInvokerFactory.builder()
+                .lambdaClient(AWSLambdaClientBuilder.defaultClient())
+                .build(Pick3PredictionService.class);
```
Comment (kalexmills) [inline chucks-pick3-ws/src/main/java/software/bytepushers/pick3/services/Pick3PredcitionServiceImpl.java]: The LambdaInvoker is using the signature of methods defined in the `Pick3PredictionService` interface to construct an RPC handler. But the interface being used is the same interface that this service implements, which is weird.

A separate interface should be used for this purpose.

File: chucks-pick3-ws/src/main/java/software/bytepushers/pick3/services/Pick3PredictionService.java [hunk 2]
```diff
@@ -1,11 +1,10 @@
...
+import com.amazonaws.services.lambda.invoke.LambdaFunction;
 import software.bytepushers.pick3.api.v1.DrawingTime;
 
 import java.time.LocalDate;
 
 public interface Pick3PredictionService {
-
-    int[][] predictPick3Plays(Integer winningNumber, LocalDate winningDrawDate, DrawingTime winningDrawTime,
-                              LocalDate futureDrawDate, DrawingTime futureDrawTime, Integer numberOfPlaysToPredict);
+    @LambdaFunction(functionName="bytepushers-chucks-pick3-lotto-system") int[][] generatePredictions(int winningNumber);
```
Comment (kalexmills) [inline chucks-pick3-ws/src/main/java/software/bytepushers/pick3/services/Pick3PredictionService.java]: This seems weird. Pick3PredictionService is simultaneously serving two roles, breaking the single-responsibility principle.

1) It is defining the signature of the AWSLambda function in order to provide the lambda function as an RPC to our web service.
1) It is defining the signature of the Pick3PredictionService which Pick3PlaysService uses to implement its call.

These two functions should be implemented as two separate interfaces / services.

File: chucks-pick3-ws/src/main/java/software/bytepushers/pick3/services/Pick3PlaysServiceImpl.java [hunk 3]
```diff
@@ -13,18 +12,15 @@
...
-    public Pick3PlaysServiceImpl(Pick3PredictionService predictionService, Pick3PlaysConfig config) {
-        this.pick3PredictionService = predictionService;
-        this.config = config;
+    public Pick3PlaysServiceImpl(Pick3PredictionService pick3PredictionService) {
+        this.pick3PredictionService = pick3PredictionService;
     }
 
     @Override
     public Pick3Plays getPick3Plays(Integer winningNumber, LocalDate winningDrawDate, DrawingTime winningDrawTime,
                                     LocalDate futureDrawDate, DrawingTime futureDrawTime) {
```
Comment (kalexmills) [inline chucks-pick3-ws/src/main/java/software/bytepushers/pick3/services/Pick3PlaysServiceImpl.java]: Why can't this be removed? It looks like it is no longer used in the tests.

Comment (pouncilt) [reply chucks-pick3-ws/src/main/java/software/bytepushers/pick3/services/Pick3PlaysServiceImpl.java]: Remove what?  Which method or line are you talking about?

File: chucks-pick3-ws/src/main/java/software/bytepushers/pick3/services/Pick3PredictionService.java (hunk 2 above)
Comment (pouncilt) [reply chucks-pick3-ws/src/main/java/software/bytepushers/pick3/services/Pick3PredictionService.java]: Yeah I kind of feel the same way.

File: chucks-pick3-ws/src/main/java/software/bytepushers/pick3/services/Pick3PredcitionServiceImpl.java (hunk 1 above)
Comment (pouncilt) [reply chucks-pick3-ws/src/main/java/software/bytepushers/pick3/services/Pick3PredcitionServiceImpl.java]: Not sure I follow.

File: chucks-pick3-ws/src/main/java/software/bytepushers/pick3/services/Pick3PlaysServiceImpl.java (hunk 3 above)
Comment (pouncilt) [reply chucks-pick3-ws/src/main/java/software/bytepushers/pick3/services/Pick3PlaysServiceImpl.java]: I am guessing you are talking about the getPick3Plays() method.  I want to keep so we can use this data for logging, tracking, and statistics.

**Pull request merged**